dev/integration_tests/web/web/index_with_flutterjs_entrypoint_loaded.html

```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE HTML>
+<!-- Copyright 2014 The Flutter Authors. All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file. -->
+<html>
+<head>
+  <meta charset="UTF-8">
+  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
+
+  <title>Integration test. App load with flutter.js and onEntrypointLoaded API</title>
+  <!-- iOS meta tags & icons -->
+  <meta name="apple-mobile-web-app-capable" content="yes">
+  <meta name="apple-mobile-web-app-status-bar-style" content="black">
+  <meta name="apple-mobile-web-app-title" content="Web Test">
+  <link rel="manifest" href="manifest.json">
+  <script>
+    // The value below is injected by flutter build, do not touch.
+    var serviceWorkerVersion = null;
+  </script>
+  <!-- This script adds the flutter initialization JS code -->
+  <script src="flutter.js" defer></script>
+</head>
+<body>
+  <script>
+    window.addEventListener('load', function(ev) {
+      // Download main.dart.js
+      _flutter.loader.loadEntrypoint({
+        serviceWorker: {
+          serviceWorkerVersion: serviceWorkerVersion,
+        }
+      });
+      // Once the entrypoint is ready, do things!
+      _flutter.loader.onEntrypointLoaded(async function(event) {
```

Comment:
It seems to me like it would be more natural to just await the promise that `loadEntrypoint` returns rather than having a disconnected callback. Is there a reason for this pattern over a standard promise await? Also, it doesn't seem like there is a mechanism to unsubscribe from these events.

Reply:
You know I'm realizing your description in the PR talks about this a lot and I didn't actually read it. Let me read through that.

Reply:
TL;DR: The `didCreateEngineInitializer` method may be called multiple times in Dev (once per hot-restart). However the promise returned by loadEntrypoint cannot be re-created, so the code breaks with a "Promise already resolved" error when you attempt to hot-restart.

The only solution I could come up with is to make an `onEntrypointLoaded` event stream, so every time that flutter tells you: "Hey, I have a fresh engine ready to be initialized" JS actually does initialize it :)

(The previous version relied on flutter's ability to auto-start if it didn't find the `didCreateEngineInitializer` callback in the DOM, by deleting it after the first load. The problem with that is that any special parameters that you passed to `initializeEngine` and `runApp` on first load, wouldn't be passed in subsequent restarts)